email-javamail/src/test/groovy/io/micronaut/email/javamail/JavaMailBodyAndAttachmentSpec.groovy

```diff
@@ -0,0 +1,148 @@
+                'javamail.properties': ['mail.smtp.host': mailHog.getContainerIpAddress(),
+                                        "mail.smtp.port": mailHog.getMappedPort(1025)]])
+    }
+
+    def "Can send an email with alternate bodies and attachments"() {
+        given:
+        EmailSender emailSender = applicationContext.getBean(EmailSender)
+        MailhogClient client = applicationContext.getBean(MailhogClient)
+
+        and:
+        String from = "sender@here.com"
+        String to = "receiver@here.com"
+        String subject = "[Javax Mail] Attachment Test" + UUID.randomUUID().toString()
+        String html = "<h1>Hola Mundo</h1>"
+        String text = "Hello world"
+        String filename = "monthlyreports.xlsx"
+        String filename2 = "weeklyreports.xlsx"
+
+        when:
+        emailSender.send(
+                Email.builder()
+                        .from(from)
+                        .to(to)
+                        .subject(subject)
+                        .body(html, text)
+                        .attachment(createSpreadsheetAttachment(filename, MediaType.MICROSOFT_EXCEL_OPEN_XML))
+                        .attachment(createSpreadsheetAttachment(filename2, MediaType.APPLICATION_OCTET_STREAM))
+        )
+
+        then:
+        conditions.eventually {
+            with(client.messages().items[0]) {
+                content.headers.Subject == subject
+                content.headers.From == from
+                content.headers.To == to
+
+                with(content.decoded()) {
+                    // Alternative body messages come first
+                    with(getBodyPart(0)) {
+                        contentType.startsWith('multipart/alternative')
+                        content.parts.find { it.contentType.startsWith(MediaType.TEXT_PLAIN) }.content == text
+                        content.parts.find { it.contentType.startsWith(MediaType.TEXT_HTML) }.content == html
+                    }
+                    // Then the attachment(s)
+                    with(getBodyPart(1)) {
+                        contentType.startsWith(MediaType.MICROSOFT_EXCEL_OPEN_XML)
+                        getHeader(HttpHeaderNames.CONTENT_DISPOSITION.toString()).head() == "attachment; filename=$filename"
+                    }
+                    with(getBodyPart(2)) {
+                        contentType.startsWith(MediaType.APPLICATION_OCTET_STREAM)
+                        getHeader(HttpHeaderNames.CONTENT_DISPOSITION.toString()).head() == "attachment; filename=$filename2"
+                    }
+                }
+            }
+        }
+    }
+
+    private Attachment createSpreadsheetAttachment(String filename, String mediaType) {
+        Attachment.builder()
+                .filename(filename)
+                .contentType(mediaType)
+                .content(SpreadsheetUtils.spreadsheet())
+                .build()
+    }
+
+    @Client('${mailhog.uri}')
+    static interface MailhogClient {
```

Comment:
annotate with `@Requires` `spec.name` to avoid bean pollution`

Reply:
resolved via https://github.com/micronaut-projects/micronaut-email/pull/45/commits/93c20359077de4309dd3983827b8abb64a20c8ea